src/core.c/IO/Socket/INET.pm6

```diff
@@ -99,10 +100,19 @@ my class IO::Socket::INET does IO::Socket {
             :$family,
         ) unless $family == nqp::const::SOCKET_FAMILY_UNIX;
 
-        # TODO: Learn what protocols map to which socket types and then determine which is needed.
+        with $localhost // $localport {
+            die 'Cannot set a source host or port for UNIX sockets'
+                if $family == nqp::const::SOCKET_FAMILY_UNIX;
+            ($localhost, $localport) = split-host-port(
+                :host($localhost),
+                :port($localport),
+                :$family
+            ) orelse fail $_;
```

Comment:
This is probably wrong, 'cus `orelse` is looser than `=`. So it's testing the `List`, which is never undefined, and `$localhost` ends up with the exception. Probably also indicates a missing test.

```
$ perl6 -e 'sub foo() { fail "bar" }; sub baz() { my ($x, $y); ($x, $y) = foo() orelse fail $_ }; dd baz()'
(Failure.new(exception => X::AdHoc.new(payload => "bar"), backtrace => Backtrace.new), Any)
$ perl6 -e 'sub foo() { fail "bar" }; sub baz() { my ($x, $y); ($x, $y) = (foo() orelse fail $_) }; dd baz()'
Failure.new(exception => X::AdHoc.new(payload => "bar"), backtrace => Backtrace.new)

```

Reply:
Another issue with this type of failure is the candidate that handles connecting doesn't do it at all. I think that belongs in a separate commit, though.